src/resolvers/resolver.ts

```diff
@@ -1,26 +1,64 @@
 import mongoose from "mongoose";
 import { Profile, User } from "../models/user";
+import bcrypt from "bcryptjs";
+import * as jwt from "jsonwebtoken";
+
+const SECRET = process.env.SECRET || "test_secret";
 
 const resolvers = {
   Query: {
-    hello: () => "Hellooo, welcome to your Graphql server",
-    async getAllUsers() {
-      const users = await User.find({});
-      return users;
-    },
-    async getAllProfiles() {
+    async getAllProfiles(_: any, _args: any, context: { userId: any }) {
+      if (!context.userId) throw new Error("Unauthorized");
       const profiles = await Profile.find({});
       return profiles;
     },
   },
   Mutation: {
-    async createUser(_: any, { email, password }: any) {
+    async createUser(
+      _: any,
+      { registerInput: { email, password, role } }: any
+    ) {
       const userExists = await User.findOne({ email: email });
       if (userExists) throw new Error("Email is taken");
-      const newUser = await User.create({ email, password });
-      return newUser;
+      const hashedPassword = await bcrypt.hash(password, 10);
+
+      const newUser = await User.create({
+        role: role || "user",
+        email: email,
+        password: hashedPassword,
+      });
+      const token = jwt.sign(
+        { userId: newUser._id, role: newUser?.role },
+        SECRET,
+        { expiresIn: "2h" }
+      );
+
+      return { token, user: newUser };
+    },
+    async loginUser(_: any, { loginInput: { email, password } }: any) {
+      const user = await User.findOne<any>({ where: { email: email } });
+
+      if (await user?.checkPass(password)) {
```

Comment:
we can also check the email before generating the token. I provided a wrong email and the right password and I was able to access the token. You can verify if the token is generated upon valid email and password.

Reply:
Chance how can we reproduce the scenario of this issue.

Reply:
@chance

how can we reproduce the scenario of this issue.